motion/thruster_interface_auv_cpp/src/thruster_interface_auv_driver.cpp

```diff
@@ -0,0 +1,138 @@
+#include "thruster_interface_auv_driver.hpp"
+
+ThrusterInterfaceAUVDriver::ThrusterInterfaceAUVDriver(){}
+
+ThrusterInterfaceAUVDriver::ThrusterInterfaceAUVDriver(
+    int I2C_BUS,
+    int PICO_I2C_ADDRESS,
+    double SYSTEM_OPERATIONAL_VOLTAGE,
+    const std::vector<int>& THRUSTER_MAPPING,
+    const std::vector<int>& THRUSTER_DIRECTION,
+    const std::vector<int>& THRUSTER_PWM_OFFSET,
+    const std::vector<int>& PWM_MIN,
+    const std::vector<int>& PWM_MAX,
+    const std::map<int, std::map<std::string, std::vector<double>>>& COEFFS
+) : I2C_BUS(I2C_BUS),
+    PICO_I2C_ADDRESS(PICO_I2C_ADDRESS),
+    //SYSTEM_OPERATIONAL_VOLTAGE(SYSTEM_OPERATIONAL_VOLTAGE), done after in the if-else
+    THRUSTER_MAPPING(THRUSTER_MAPPING),
+    THRUSTER_DIRECTION(THRUSTER_DIRECTION),
+    THRUSTER_PWM_OFFSET(THRUSTER_PWM_OFFSET),
+    PWM_MIN(PWM_MIN),
+    PWM_MAX(PWM_MAX),
+    COEFFS(COEFFS)
+{
+
+    // Convert SYSTEM_OPERATIONAL_VOLTAGE passed as argument to assign the integer field variable [one is double, field var is int]
+    if (SYSTEM_OPERATIONAL_VOLTAGE < 11.0) {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 10;
+    } else if (SYSTEM_OPERATIONAL_VOLTAGE < 13.0) {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 12;
+    } else if (SYSTEM_OPERATIONAL_VOLTAGE < 15.0) {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 14;
+    } else if (SYSTEM_OPERATIONAL_VOLTAGE < 17.0) {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 16;
+    } else if (SYSTEM_OPERATIONAL_VOLTAGE < 19.0) {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 18;
+    } else {
+        this->SYSTEM_OPERATIONAL_VOLTAGE = 20;
+    }
+
+    // Open the I2C bus
+    std::string i2c_filename = "/dev/i2c-" + std::to_string(I2C_BUS);
+    bus_fd = open(i2c_filename.c_str(), O_RDWR);
+    if (bus_fd < 0) {
+        std::cerr << "ERROR: Failed to open I2C bus " << I2C_BUS << std::endl;
+    }
+}
+
+ThrusterInterfaceAUVDriver::~ThrusterInterfaceAUVDriver() {
+    if (bus_fd >= 0) {
+        close(bus_fd);
+    }
+}
+
+std::vector<int16_t> ThrusterInterfaceAUVDriver::interpolate_forces_to_pwm(const std::vector<double>& thruster_forces_array) {
+    // Convert Newtons to Kg (since the thruster datasheet is in Kg)
+    std::vector<double> forces_in_kg(thruster_forces_array.size());
+    for (size_t i = 0; i < thruster_forces_array.size(); ++i) {
+        forces_in_kg[i] = thruster_forces_array[i] / 9.80665;
+    }
```

Comment:
i put it in the local scope of the driver class as **static constexpr** directly in the .hpp class, instead of declaring it and defining it in the .cpp, since it's a single line of code. Is it okay? Compiling is compiling and working too.